src/main/java/io/kestra/plugin/ai/tool/AIAgent.java

```diff
@@ -0,0 +1,215 @@
+import io.kestra.core.utils.ListUtils;
+import io.kestra.plugin.ai.AIUtils;
+import io.kestra.plugin.ai.domain.*;
+import io.swagger.v3.oas.annotations.media.Schema;
+import jakarta.validation.constraints.NotNull;
+import lombok.AccessLevel;
+import lombok.Builder;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.experimental.SuperBuilder;
+
+import java.util.List;
+import java.util.Map;
+
+import static io.kestra.core.utils.Rethrow.throwFunction;
+
+@Getter
+@SuperBuilder
+@NoArgsConstructor
+@Plugin(
+    examples =  {
+        @Example(
+            title = "Call a AI agent as a tool",
+            full = true,
+            code = {
+                """
+                    id: ai-agent-with-agent-tools
+                    namespace: company.ai
+
+                    inputs:
+                      - id: prompt
+                        type: STRING
+                        defaults: |
+                          Each flow can produce outputs that can be consumed by other flows. This is a list property, so that your flow can produce as many outputs as you need.
+                          Each output needs to have an id (the name of the output), a type (the same types you know from inputs e.g. STRING, URI or JSON) and value which is the actual output value that will be stored in internal storage and passed to other flows when needed.
+                    tasks:
+                      - id: ai-agent
+                        type: io.kestra.plugin.ai.agent.AIAgent
+                        provider:
+                          type: io.kestra.plugin.ai.provider.GoogleGemini
+                          modelName: gemini-2.5-flash
+                          apiKey: "{{ kv('GEMINI_API_KEY') }}"
+                        systemMessage: You are an AI agent, summarize the user message then translate it in french using the provided tool.
+                        prompt: "{{inputs.prompt}}"
+                        tools:
+                          - type: io.kestra.plugin.ai.tool.Agent
+                            description: An AI agent expert on translation
+                            systemMessage: You are an expert system on translation
+                            provider:
+                              type: io.kestra.plugin.ai.provider.GoogleGemini
+                              modelName: gemini-2.5-flash-lite
+                              apiKey: "{{ kv('GEMINI_API_KEY') }}\""""
+            }
+        ),
+    }
+)
+@JsonDeserialize
+@Schema(
+    title = "Call a AI Agent as a tool",
+    description = """
+        This tool allows an LLM to call a AI Agent.
+        The agent must have a name and a description so the LLM can understand what it does to decide if it needs to call it."""
+)
+public class Agent extends ToolProvider {
+    private static final String TOOL_DESCRIPTION = "This tool allows to call an AI agent named '%s'.";
+
+    @Schema(
+        title = "Agent name",
```

Comment:
```suggestion
        name = "Agent name",
```
if it's a `name`, should we call the property a name? You seem to refer it everywhere as name, so maybe worth renaming?

Reply:
Here it's the title of the OpenAPI schema not the name of the property which is `name`.